test/blackbox/common/DDSBlackboxTestsDiscovery.cpp

```diff
@@ -52,4 +61,110 @@ TEST(DDSDiscovery, IgnoreParticipantFlags)
     p3.init();
     EXPECT_TRUE(p1.wait_participant_discovery());
     EXPECT_TRUE(p3.wait_participant_discovery());
-}
\ No newline at end of file
+}
+
+/**
+ * This test checks that adding servers to the Discovery Server list results in discovering those participants.
+ * It does so by:
+ *    1. Creating two servers and one client that is only connected to one server. At this point check discovery
+ *       state.
+ *    2. Then, connect the client to the other server and check discovery again.
+ *    3. Finally connect the two servers by adding one of them to the others list
+ */
+TEST(DDSDiscovery, AddDiscoveryServerToList)
+{
+    using namespace eprosima::fastdds::dds;
+    using namespace eprosima::fastrtps::rtps;
+
+    /* Get random port from the environment */
+    char* value = nullptr;
+    value = std::getenv("W_UNICAST_PORT_RANDOM_NUMBER");
```

Comment:
Fast DDS provides a method to read environment variables get_env(). It can be found in private header file utils/SystemInfo.hpp. This method is tested for all supported platforms. I suggest using it so we have only to maintain one method in case we start supporting new platforms in the future.